converter/text.py

```diff
@@ -324,3 +326,30 @@ def text_transformation(fig_text):
         return {"MSAttributedStringTextTransformAttribute": TextCase[fig_text["textCase"]]}
     else:
         return {}
+
+
+def font_features(fig_text: dict) -> List[OTFeature]:
+    sketch_features = []
+    unsupported_features = []
+    for f in fig_text.get("toggledOnOTFeatures", []):
+        try:
+            fid, on, off = FEATURE_MAPPINGS[f.lower()]
+            sketch_features.append(
+                OTFeature(CTFeatureSelectorIdentifier=on, CTFeatureTypeIdentifier=fid)
+            )
+        except KeyError:
+            unsupported_features.append(f)
+
+    for f in fig_text.get("toggledOffOTFeatures", []):
```

Comment:
Sounds like these 2 loops could be refactored into a small function that gets the property to check (`toggledOnOTFeatures` or `toggleOffOTFeatures`) and the the value for `CTFeatureSelectorIdentifier` (`on` or `off`) and return the list of unsupported features.

Reply:
Yeah, I tried and didn't like it. You have to pass two parameters to it, the array of fig_text and the index of the on/off value. Additionally, you have to return two lists (sketch_features and unsupported_features) and append them. It ended up looking worse IMHO.

Reply:
Oh, right. Forget it then :homerdissapears: